File: src/b3085.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int check(vector<string> bom, int n)
{
	int max = 1;
	int count = 1;

	for (int i = 0; i < n; i++)
	{
		count = 1;
		for (int j = 1; j < n; j++)
		{
			if (bom[i][j] == bom[i][j - 1]) count++;
			else count = 1;
			if (max < count) max = count;
		}
		count = 1;
		for (int j = 1; j < n; j++)
		{
			if (bom[j][i] == bom[j - 1][i]) count++;
			else count = 1;
			if (max < count) max = count;
		}
	}
	return max;

}
int main(void)
{
	int n;
	int tmp, max = 0;

	cin >> n;
	vector<string> bom(n);

	for (int i = 0; i < n; i++)
	{
		cin >> bom[i];
	}

	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			if (j + 1 < n)		//오른쪽
			{
				swap(bom[i][j], bom[i][j + 1]);
				tmp = check(bom, n);
				if (max < tmp) max = tmp;
				swap(bom[i][j], bom[i][j + 1]);
			}
			if (i + 1 < n)		//아래
			{
				swap(bom[i][j], bom[i + 1][j]);
				tmp = check(bom, n);
				if (max < tmp) max = tmp;
				swap(bom[i][j], bom[i + 1][j]);
			}
		}
	}

	cout << max << endl;
}

File: src/b3085.py
import sys

def check(board):
    count=1
    maxi=1
    for i in range(0,n):
        count=1
        for j in range(1,n):
            if board[i][j]==board[i][j-1]:
                count+=1
            else:
                count=1
            if(count>maxi):
                maxi=count
        count=1
        for j in range(1,n):
            if board[j][i]==board[j-1][i]:
                count+=1
            else:
                count=1
            if(count>maxi):
                maxi=count
    
    return maxi

input=sys.stdin.readline
n=int(input())
board=[list(input().rstrip('\n')) for _ in range(n)]
res=0
tmp=0
for i in range (0,n):
    for j in range (0,n):
        if i+1<n:
            board[i+1][j],board[i][j]=board[i][j],board[i+1][j]
            tmp=check(board[:])
            if tmp>res:
                res=tmp
            board[i+1][j],board[i][j]=board[i][j],board[i+1][j]
        
        if j+1<n:
            board[i][j+1],board[i][j]=board[i][j],board[i][j+1]
            tmp=check(board[:])
            if tmp>res:
                res=tmp
            board[i][j+1],board[i][j]=board[i][j],board[i][j+1]
        

print(res)
